tock/hours/tests/test_forms.py

```diff
@@ -1,114 +1,123 @@
-import os
-import shutil
-from unittest.mock import Mock, patch
 import datetime
 
-from django.core.exceptions import ValidationError
-from django.conf import settings
 from django.test import TestCase
 from django.contrib.auth import get_user_model
 
 import hours.models
 import projects.models
 
-from hours.forms import TimecardForm, TimecardObjectForm, TimecardFormSet, TimecardInlineFormSet
+from hours.forms import TimecardForm, TimecardObjectForm, TimecardFormSet
 
 
 class TimecardFormTests(TestCase):
-  fixtures = ['projects/fixtures/projects.json', 'tock/fixtures/dev_user.json']
-
-  def setUp(self):
-    self.reporting_period = hours.models.ReportingPeriod.objects.create(
-        start_date=datetime.date(2015, 1, 1),
-        end_date=datetime.date(2015, 1, 7),
-        working_hours=40)
-    self.user = get_user_model().objects.get(id=1)
-    self.project_1 = projects.models.Project.objects.get(name="openFEC")
-    self.project_2 = projects.models.Project.objects.get(name="Peace Corps")
-
-  def tearDown(self):
-    hours.models.ReportingPeriod.objects.all().delete()
-    hours.models.Timecard.objects.all().delete()
-    projects.models.Project.objects.all().delete()
-    hours.models.TimecardObject.objects.all().delete()
-
-  def test_valid_form(self):
-    form_data = {'user': self.user, 'reporting_period': self.reporting_period}
-    form = TimecardForm(form_data)
-    self.assertTrue(form.is_valid())
+    fixtures = [
+        'projects/fixtures/projects.json', 'tock/fixtures/dev_user.json']
+
+    def setUp(self):
+        self.reporting_period = hours.models.ReportingPeriod.objects.create(
+            start_date=datetime.date(2015, 1, 1),
+            end_date=datetime.date(2015, 1, 7),
+            working_hours=40)
+        self.user = get_user_model().objects.get(id=1)
+        self.project_1 = projects.models.Project.objects.get(name="openFEC")
+        self.project_2 = projects.models.Project.objects.get(
+            name="Peace Corps")
+
+    def tearDown(self):
```

Comment:
The base test class should wrap every test method in a transaction which is rolled back afterwards, so we shouldn't have to manually clear tables. I would hope this is already part of the Django testing harness, but if not, let's add it.